public/components/data_connections/components/access_control_tab.tsx

```diff
@@ -0,0 +1,205 @@
+    );
+  };
+
+  const renderEditAccessControlDetails = () => {
+    return (
+      <EuiFlexGroup direction="column">
+        <QueryPermissionsFlexItem
+          roles={roles}
+          selectedRoles={selectedQueryPermissionRoles}
+          setSelectedRoles={setSelectedQueryPermissionRoles}
+          selectedRadio={queryPermissionRadioSelected}
+          setSelectedRadio={setQueryRadioIdSelected}
+          radios={queryRadios}
+        />
+        <AccelerationPermissionsFlexItem
+          roles={roles}
+          selectedRoles={selectedAccelerationPermissionRoles}
+          setSelectedRoles={setSelectedAccelerationPermissionRoles}
+          selectedRadio={accelerationPermissionRadioSelected}
+          setSelectedRadio={setAccelerationRadioIdSelected}
+          radios={accelerationRadios}
+        />
+      </EuiFlexGroup>
+    );
+  };
+
+  const saveChanges = () => {
+    setFinalAccelerationPermissionsPlaceHolder(selectedAccelerationPermissionRoles);
+    setFinalQueryPermissionsPlaceHolder(selectedQueryPermissionRoles);
+    setMode('view');
+  };
+
+  return (
+    <>
+      <EuiSpacer />
+      <AccessControlCallout />
+      <EuiSpacer />
+      <EuiPanel>
+        <EuiFlexGroup direction="row">
+          <EuiFlexItem>
+            <EuiText size="m">
+              <h2 className="panel-title">Access Control</h2>
+              Control which OpenSearch users have access to this data source.
+            </EuiText>
+          </EuiFlexItem>
+
+          <EuiFlexItem grow={false}>
+            <EuiButton
+              data-test-subj="createButton"
+              onClick={
+                mode === 'view'
+                  ? () => {
+                      setMode('edit');
+                    }
+                  : () => {
+                      setMode('view');
+                    }
+              }
+              fill={mode === 'view' ? true : false}
+            >
+              {mode === 'view' ? 'Edit' : 'Cancel'}
+            </EuiButton>
+          </EuiFlexItem>
+        </EuiFlexGroup>
+        <EuiHorizontalRule />
+        {mode === 'view' ? renderViewAccessControlDetails() : renderEditAccessControlDetails()}
+      </EuiPanel>
+      <EuiSpacer />
+      {mode === 'edit' ? (
+        <EuiBottomBar affordForDisplacement={false}>
+          <EuiFlexGroup justifyContent="flexEnd">
+            <EuiFlexItem grow={false}>
+              <EuiButtonEmpty
+                onClick={() => {
+                  setMode('view');
+                }}
+                color="ghost"
+                size="s"
+                iconType="cross"
+              >
+                Discard change(s)
+              </EuiButtonEmpty>
+            </EuiFlexItem>
+            <EuiFlexItem grow={false}>
+              <EuiButton onClick={saveChanges} size="s" iconType="check" fill>
+                Save
+              </EuiButton>
+            </EuiFlexItem>
+          </EuiFlexGroup>
+        </EuiBottomBar>
+      ) : null}
```

Comment:
Minor can be changed to `mode === 'edit' && <view/>` no need to use null.

Reply:
Done in: [fc96304](https://github.com/opensearch-project/dashboards-observability/pull/992/commits/fc96304362b085696c2e5df8e7c638b69c142180), thanks!